daemon/src/wallet.rs

```diff
@@ -109,6 +117,26 @@ impl Actor<ElectrumBlockchain, sled::Tree> {
         let client = bdk::electrum_client::Client::new(electrum_rpc_url)
             .context("Failed to initialize Electrum RPC client")?;
 
+        let network_hash = constants::genesis_block(ext_priv_key.network).block_hash();
+        // This needs to be converted to a BlockHash first, as Sha256d hashes are displayed backward
+        let rpc_hash = BlockHash::from_byte_iter(
+            client
+                .server_features()?
+                .genesis_hash
+                .iter()
+                .map(|c| Ok(*c)),
+        )
+        .context("Invalid genesis block hash returned by electrum RPC")?;
```

Comment:
❓ Doesn't this work:

```suggestion
        let rpc_hash = BlockHash::from_slice(&client.server_features()?.genesis_hash)?;
```

Reply:
Thanks, I didn't know about this method!

Reply:
This turns out to not reverse the bytes for us, so we must first do this ourselves. This is fewer lines but it may be slightly less clear